### PR TITLE
Update to Go 1.20 (and trim the dependency tree)

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Chore -- upgrade to 1.20

### What this PR does / why we need it:

Updates ARO to build on Go 1.20. Also updates a few very old deps (I'll pull this out into a different one)

### CODE REVIEW

**Review by mociarain — COMMENTED** [review]
Just a small comment on phrasing. I need to re-install Go so I can test this fully later on today

File: docs/prepare-your-dev-environment.md [hunk 1]
```diff
@@ -4,48 +4,50 @@ This document goes through the development dependencies one requires in order to
 
 ## Software Required
 
-1. Install [Go 1.18](https://golang.org/dl) or later, if you haven't already.
-   1. After downloading follow the [Install instructions](https://go.dev/doc/install), replacing the tar archive with your download.
-   1. Append `export PATH="${PATH}:/usr/local/go/bin"` to your shell's profile file.
+1.  Install [Go 1.20](https://golang.org/dl) or later, if you haven't already.
```
[the remark this answers is not in the document]
Comment (tsatam) [reply docs/prepare-your-dev-environment.md:7]: We should test, but I hope that getting past Go 1.19 should resolve this issue. As far as I'm aware, the only problem we historically had with "later versions" was Go 1.19 changing whitespace in generated files we commit into source control. As far as I am aware it's not really the norm to limit your installed Go version to the specific version of Go defined in a project's go.mod, newer versions should generally be compatible with older codebases outside of niche issues like ours. 

If you experienced other issues than this one, I'd like to see if we could resolve those directly, and allow us working on the repo to just do so with the latest version of Go installed.

[the remark this answers is not in the document]
Comment (mociarain) [reply docs/prepare-your-dev-environment.md:7]: For reference what happened was I got a bunch of small auto-generated linting changes from the generate and build-all `make` targets when I was using 1.21.x

I'm new to Go so happy to defer to those with more experience

[the remark this answers is not in the document]
Comment (tsatam) [reply docs/prepare-your-dev-environment.md:7]: That is the problem I was describing, so hopefully getting the codebase past 1.19 will eliminate that problem for you. This PR will eventually need to include all of those changes in order to pass CI checks.

**Comment by hawkowl** [comment]
supersceded by https://github.com/Azure/ARO-RP/pull/3385

**Pull request closed without merging**